File: examples/csharp/documentation/DataSources.cs
// DOCS_BLOCK_START:datasources
using System;
using System.IO;
using System.Collections.Generic;
using System.Linq;
using Constructs;
using HashiCorp.Cdktf;
using aws.Provider;
using aws.DataAwsRegion;

namespace Examples
{
    class DataSourceStack : TerraformStack
    {
        public DataSourceStack(Construct scope, string name) : base(scope, name)
        {

            new AwsProvider(this, "aws", new AwsProviderConfig {
                Region = "eu-central-1"
            });

            DataAwsRegion region = new DataAwsRegion(this, "region", new DataAwsRegionConfig {
                Name = "eu-central-1"
            });
        }
    }
}
// DOCS_BLOCK_END:datasources

File: examples/go/documentation/datasources.go
package main

// DOCS_BLOCK_START:datasources
import (
	"github.com/aws/constructs-go/constructs/v10"
	"github.com/aws/jsii-runtime-go"
	"github.com/hashicorp/terraform-cdk-go/cdktf"
	"github.com/hashicorp/terraform-cdk/examples/go/documentation/generated/hashicorp/aws/dataawsregion"
	aws "github.com/hashicorp/terraform-cdk/examples/go/documentation/generated/hashicorp/aws/provider"
)

func NewDatasourcesStack(scope constructs.Construct, name string) cdktf.TerraformStack {
	stack := cdktf.NewTerraformStack(scope, &name)

	aws.NewAwsProvider(stack, jsii.String("aws"), &aws.AwsProviderConfig{
		Region: jsii.String("eu-central-1"),
	})

	dataawsregion.NewDataAwsRegion(stack, jsii.String("region"), &dataawsregion.DataAwsRegionConfig{
		Name: jsii.String("eu-central-1"),
	})

	return stack
}

// DOCS_BLOCK_END:datasources
